packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1742,6 +1805,19 @@ export class KeyringController extends BaseController<
     );
   }
 
+  /**
+   * Get the keyring by id.
+   *
+   * @param keyringId - The id of the keyring.
+   * @returns The keyring.
+   */
+  #getKeyringById(keyringId: string): EthKeyring<Json> | undefined {
+    const index = this.state.keyringsMetadata.findIndex(
```

Comment:
Open question: Do we want to add an `assert` here? since both arrays should be "in-sync"? (Talking about this one: https://github.com/MetaMask/utils/blob/main/src/assert.ts#L81-L86)

IDK if we use this pattern elsewhere already, maybe @mikesposito would have an opinion on this?

```suggestion
    assert(
      this.state.keyringsMetadata.length === this.#keyrings.length,
      "Keyrings and keyrings metadata length mismatch
    );
    const index = this.state.keyringsMetadata.findIndex(
```

Reply:
`assert` is a nice function that I've personally used to avoid typescript assertions - though while it's true that these should always have the same length, I'm not sure it's necessary to check it here because there is no assertion needed and `findIndex` already defaults to `-1`, which would return `undefined`

Reply:
Yup that's true, I did not check the return type, but we have an `| undefined`, so that's a possible value too.

Then we should definitely not add a `assert` here. Thanks!